core/domain/classifier_services.py

```diff
@@ -347,6 +348,75 @@ def mark_training_job_complete(job_id):
                                            feconf.TRAINING_JOB_STATUS_COMPLETE)
 
 
+def mark_training_job_failed(job_id):
+    """Updates the training job's status to failed.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_job_status(job_id,
+                                           feconf.TRAINING_JOB_STATUS_FAILED)
+
+
+def mark_training_job_pending(job_id):
+    """Updates the training job's status to pending.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_job_status(job_id,
+                                           feconf.TRAINING_JOB_STATUS_PENDING)
+
+
+def update_failed_jobs(training_jobs):
+    """Updates status of job_models to failed.
+
+    Args:
+        training_jobs: list. List of jobs to be updated as failed.
+    """
+    for training_job in training_jobs:
+        mark_training_job_failed(training_job.job_id)
+
+
+def fetch_next_job():
+    """Gets next job model in the job queue.
+
+    Returns:
+        ClassifierTrainingJobModel. Domain object of the next training Job.
+    """
+    classifier_training_jobs = []
+    cursor = None
+    valid_jobs = []
+    failed_jobs = []
+
+    while len(valid_jobs) == 0:
```

Comment:
1st question: Let say we do nothing. Just keep record of how many jobs actually fail? If they are very severe then we retry the jobs or email admin. If it happens frequently then we probably need to retry them and improve our approach to make failing less frequent. But as of now I am not sure which one is the best. We can take more conclusive decision once its in production, I think. What do you think?

2nd question: the order as of now seems FIFO to me. We start checking jobs in order in which they were created (query on created_on field). Then we start to find a valid job (NEW job) and if we encounter any pending job with TTL exceeded we mark it as failed. If TTL is not exceeded then ignore the job. We will come across that job later when VM makes another request for job. This seems FIFO to me. I don't think we would need last_checked_time field for FIFO ordering. Why can't we use created_on field directly?